File: Data Structures & Algorithms/jump-game-ii/submission-3.cpp
class Solution {
public:
    int dfs(int i,vector<int>& nums){
        if(i>= nums.size()) return 100000;
        if(i==nums.size()-1) return 0;
        int res =1000000;
        for(int j=1;j<= nums[i];j++){
            res=min(res,dfs(i+j,nums)+1);
        }
        return res ;
    }
    int jump(vector<int>& nums) {
        return dfs(0,nums);
    }
};

File: Data Structures & Algorithms/jump-game-vii/submission-3.py
class Solution:
    def canReach(self, s: str, minJump: int, maxJump: int) -> bool:
        n=len(s)
        memo={}
        if s[n-1]=='1' :
            return False
        if maxJump == 0 :
            return False
        def jump(i) :
            if i==n-1 :
                return True
            if s[i]!='0' :
                return False
            if i in memo :
                return memo[i]
            for j in range(min(i+maxJump,n-1) ,i+minJump-1,-1):
                if jump(j):
                    memo[i]= True
                    return True
            memo[i] = False
            return False
        for j in range(min(maxJump,n-1) ,minJump-1,-1):
            if jump(j) :
                return True
        return False
    